src/eval.c

```diff
@@ -3605,6 +3618,294 @@ get_lit_string_tv(char_u **arg, typval_T *rettv, int evaluate)
     return OK;
 }
 
+    static int
+is_expected_closing(char_u expected, char_u exclude1, char_u exclude2, char_u *template, int *is_skipping_needed)
+{
+    if (*template == exclude1 && *(template + 1) == exclude2)
+    {
+	*is_skipping_needed = TRUE;
+        return FALSE;
+    }
+
+    *is_skipping_needed = FALSE;
+    return *template == expected;
+}
+
+    static int
+is_closing_quote(char_u *template, int is_literal_string, int *is_skipping_needed)
```

Comment:
I intended something like this (Not tested):

```C
    static int
is_closing_quote(char_u *template, int is_literal_string, int *is_skipping_needed)
{
    char_u  quote = is_literal_string ? '\'' : '"';

    if (is_escaped_quote(is_literal_string, template))
    {
	*is_skipping_needed = TRUE;
	return FALSE;
    }

    *is_skipping_needed = FALSE;
    return *template == quote;
}
```

Isn't this simple enough even `is_expected_closing()` is merged into this?

Reply:
Thanks, I'll use it!